pkg/core/resources/store/index.go

```diff
@@ -0,0 +1,14 @@
+package store
+
+import (
+	"context"
+
+	"github.com/kumahq/kuma/pkg/core/resources/model"
+)
+
+type IndexStore interface {
+	Create(context.Context, model.Resource, ...CreateOptionsFunc) error
+	Update(context.Context, model.Resource, ...UpdateOptionsFunc) error
+	Delete(context.Context, model.Resource, ...DeleteIndexOptionsFunc) error
+	List(context.Context, model.ResourceList, ...ListIndexOptionsFunc) error
+}
```

Comment:
What is the motivation for introducing a new interface for the store?

Reply:
When the parent node is deleted, the data of the child nodes needs to be deleted, so we store the keys of all child nodes under the parent node

Reply:
I might be confused, but AFAIU the reason for the new interface is the owner part here?
```
type DeleteIndexOptions struct {
	Name string
	Mesh string
	Type core_model.ResourceType

	Owner struct {
		Name string
		Mesh string
		Type core_model.ResourceType
	}
}
```

I see that in etcd store implementation when we delete the resource we first retrieve it. Could we just extract the owner there instead of requiring users of IndexStore to provide it?

Reply:
When I store an object A whose owner is B, at this time, I can save A’s Owner B as an attribute of A to object A. This storage will not modify the attributes of B. If I need to delete object B now , according to the current implementation, the owner of A is B. When the B object is deleted, A also needs to be deleted, but for the value of the B object, I can't get B to have a child node A, so I need to add a new Index structure to support this relation

Reply:
ok, but how would the client of this IndexStore use it? Wouldn't this index be essentially something internal?

btw. `Type core_model.ResourceType` in Opts is not needed. You always pass a real object to Get, List etc. and you can then call `model.Resource#Descriptor().Name`